src/effects/backends/builtin/echoeffect.cpp

```diff
@@ -236,16 +250,17 @@ void EchoEffect::processChannel(
             pGroupState->ping_pong = 0;
         }
     }
-
-    // The ramping of the send parameter handles ramping when enabling, so
-    // this effect must handle ramping to dry when disabling itself (instead
-    // of being handled by EngineEffect::process).
+    pGroupState->prev_send = send_current;
     if (enableState == EffectEnableState::Disabling) {
-        SampleUtil::applyRampingGain(pOutput, 1.0, 0.0, engineParameters.samplesPerBuffer());
-        pGroupState->delay_buf.clear();
         pGroupState->prev_send = 0;
-    } else {
-        pGroupState->prev_send = send_current;
+        const SINT delayBufferSize = pGroupState->delay_buf.size();
+        // Calculate if the delayline-buffer is approx. zero/empty.
+        const float avgSampleEnergy = averageSampleEnergy(pGroupState->delay_buf);
+        // If echo tail fully faded
+        if (avgSampleEnergy < (0.00001f / delayBufferSize)) {
```

Comment:
why does the energy depend on the buffer size? The average shouldn't be influenced by the sample size of the dataset. 

If the concern is that the fading tail influences the current average, simply don't sample it. Only sample as much as you need:

Reply:
The average sample energy depends on the buffer-size since it is computed from the buffer (average over buffer length, the average over the next buffer as well, or only half the buffer, would be numerically different). Thus, `avgSampleEnergy` implicitly depends on the buffer length.

Before your previous review, the `averageSampleEnergy` would not return an average but the summed Sample Energy over the buffer, this was simply bad code and is now fixed. To retain the exactly same threshold which was test-driven by me as before, the `0.00001f` magic number now needs to be deleted by the `delayBufferSize`. Of course I can also shift the `0.00001f`  a bit by making it 100x smaller or so and call it a day :--) let me know

Reply:
> Thus, avgSampleEnergy implicitly depends on the buffer length.

Well, now that we're only looking at the samples at the beginning of the delay buffer (exactly the samples that feedback into the effect to be exact) the size shouldn't anymore, right? Am I making sense here?

Besides I think `.first(engineParameters.sampleRate())` is probably wrong, my mistake. I meant `.first(<how big the current playback buffer is>)`...

Reply:
Mhm I am not sure I follow sorry, I think the average is calculated over the whole delay buffer. But we can in any case remove the division by the delay buffer length in the treshhold check. This is just numerics:--)